js/apps/system/_admin/aardvark/APP/react/src/views/views/searchAliasView/useUpdateAliasViewProperties.ts

```diff
@@ -0,0 +1,160 @@
+import { ViewPropertiesType } from "./useFetchViewProperties";
+
+export const useUpdateAliasViewProperties = ({
+  setChanged
+}: {
+  setChanged: (changed: boolean) => void;
+}) => {
+  const onSave = async ({
+    view,
+    initialView
+  }: {
+    view: ViewPropertiesType;
+    initialView: ViewPropertiesType;
+  }) => {
+    try {
+      const isNameChanged = initialView.name && view.name !== initialView.name;
+      let isError = false;
+      if (isNameChanged) {
+        isError = await putRenameView({
+          initialName: initialView.name,
+          name: view.name
+        });
+      }
+
+      if (!isError) {
+        await patchViewProperties({
+          view,
+          isNameChanged,
+          initialView,
+          setChanged
+        });
+      }
+    } catch (e) {
+      window.arangoHelper.arangoError(
+        "Failure",
+        `Got unexpected server response: ${e.message}`
+      );
+    }
+  };
+
+  return { onSave };
+};
+
+const putRenameView = async ({
+  initialName,
+  name
+}: {
+  initialName: string;
+  name: string;
+}) => {
+  let isError = false;
+  const route = getApiRouteForCurrentDB();
+
+  const result = await route.put(`/view/${initialName}/rename`, {
+    name: name
+  });
+
+  if (result.body.error) {
+    window.arangoHelper.arangoError(
+      "Failure",
+      `Got unexpected server response: ${result.body.errorMessage}`
+    );
+    isError = true;
+  }
+  return isError;
+};
+async function patchViewProperties({
+  view,
+  isNameChanged,
+  initialView,
+  setChanged
+}: {
+  view: ViewPropertiesType;
+  isNameChanged: string | boolean;
+  initialView: ViewPropertiesType;
+  setChanged: (changed: boolean) => void;
+}) {
+  const path = `/view/${view.name}/properties`;
+  const result = await patchProperties({ view, path, initialView });
+  window.arangoHelper.hideArangoNotifications();
+
+  if (result.body.error) {
+    window.arangoHelper.arangoError(
+      "Failure",
+      `Got unexpected server response: ${result.body.errorMessage}`
+    );
+  } else {
+    setChanged(false);
+    window.sessionStorage.removeItem(`${initialView.name}-changed`);
+    window.sessionStorage.removeItem(`${initialView.name}`);
+
+    if (!isNameChanged) {
+      await mutate(path);
+    } else {
+      let newRoute = `#view/${view.name}`;
+      window.App.navigate(newRoute, {
+        trigger: true,
+        replace: true
+      });
+    }
+
+    window.arangoHelper.arangoNotification(
+      "Success",
+      `Updated View: ${view.name}`
+    );
+  }
+}
+
+async function patchProperties({
+  view,
+  path,
+  initialView
+}: {
+  view: ViewPropertiesType;
+  initialView: ViewPropertiesType;
+  path: string;
+}) {
+  const route = getApiRouteForCurrentDB();
+
+  window.arangoHelper.arangoMessage(
+    "Saving",
+    `Please wait while the view is being saved. This could take some time for large views.`
+  );
+  const { changesAdded, changesDeleted } = getUpdatedIndexes({
```

Comment:
Maybe we should consider renaming this as follows
Instead of `changesAdded` to `addedChanges`

Similar for the deleted one. Let's follow the verb+entity/noun pattern